### PR TITLE
WIP : Feature/notifications

### PR DESCRIPTION
**Pull request by btry**
This is WIP, sugestions welcome

TODO
- more explanations
- run the example code to ensure it actually works

### CODE REVIEW

File: source/plugins/notifications.rst [hunk 1]
```diff
@@ -0,0 +1,231 @@
...
+
+Send email notifications to users when some event occurs
+
+
+Add events on a notification in GLPI core
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+If the plugin needs to create a new event for a notification which belongs to GLPI core.
+
+The folowing example creates an event `plugin_exemple` for Ticket notifications.
```
Comment (orthagh) [inline source/plugins/notifications.rst]: exemple -> example

File: source/plugins/notifications.rst [hunk 2]
```diff
@@ -0,0 +1,231 @@
...
+
+   GLPI also supports hook methods.
+
+
+Add data on a event in GLPI core
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+If the plugin needs to create new placeholders in a notification which belongs to GLPI core.
+
+The following example defines new data placeholders for Ticket notrifications.
```
Comment (orthagh) [inline source/plugins/notifications.rst]: typo: notrifications

File: source/plugins/notifications.rst [hunk 3]
```diff
@@ -0,0 +1,231 @@
...
+
+   In the notification creation, the code uses ther constant `PluginExampleNotificationTargetCertificate::EVENT_EXPIRATION`. Don't forget to define it.
+   
+Add the method getTags() to enumerate the tags available in the notification. It is used to display a list of tags when an adminitrators creates or edit a notificatoin template translation.
+
+.. code-block:: php
+
+   <?php
+   // File inc/notificationtargetcertificate.class.php
+   class PluginExampleNotificationTargetCertificate extends CommonDBTM
```
Comment (orthagh) [inline source/plugins/notifications.rst]: extends NotificationTarget

File: source/plugins/notifications.rst [hunk 4]
```diff
@@ -0,0 +1,231 @@
...
+         $this->addTagToList(array('tag'    => $tag,
+               'label'  => $label,
+               'value'  => true,
+               'events' => NotificationTarget::TAG_FOR_ALL_EVENTS));
+      }
+   }
+
+   }
+
+.. code-block:: php
```
Comment (orthagh) [inline source/plugins/notifications.rst]: why double declaration of the same class ?

**Comment by trasher** [comment]
PR stalled for a while, to be closed?

**Comment by btry** [comment]
Don't close please, I'll re-add it to my tasks queue.

**Comment by btry** [comment]
I rebased the branch and solved conflicts. 

content of ```notification.rst``` was moved to ```notifications modes.rst``` (this is not obvious by reading the diff).

It seems my contribution does not duplicates the already available doc about notifications.

**Comment by trasher** [comment]
Stalled for a while, closing.

**Pull request closed without merging**